HpToolsLauncher/Launcher.cs

```diff
@@ -507,7 +513,7 @@ IAssetRunner CreateRunner(TestStorageType runType, JavaProperties ciParams)
                         mobileinfo = _ciParams["mobileinfo"];
                     }
 
-                    runner = new FileSystemTestsRunner(validTests, timeout, pollingInterval, perScenarioTimeOutMinutes, ignoreErrorStrings, jenkinsEnvVariables, mcConnectionInfo, mobileinfo);
+                    runner = new FileSystemTestsRunner(validTests, timeout, uftRunMode, pollingInterval, perScenarioTimeOutMinutes, ignoreErrorStrings, jenkinsEnvVariables, mcConnectionInfo, mobileinfo);
```

Comment:
It will happen only if you don't call it.
You should call it if and only if you have the required parameter if not then you can use the other one.
Actually it will be a  great step to start decoupling this step.